src/components/layout/Navbar/index.tsx

```diff
@@ -1,18 +1,39 @@
 import { config } from '@/lib';
 import type { PrivateProfile } from '@/lib/types/apiResponses';
 import LightModeLogo from '@/public/assets/acm-logos/general/light-mode.png';
+import ACMIcon from '@/public/assets/icons/acm-icon.svg';
+import CalendarIcon from '@/public/assets/icons/calendar-icon.svg';
+import LeaderboardIcon from '@/public/assets/icons/leaderboard-icon.svg';
 import ProfileIcon from '@/public/assets/icons/profile-icon.svg';
 import ShopIcon from '@/public/assets/icons/shop-icon.svg';
 import Image from 'next/image';
 import Link from 'next/link';
-import { memo } from 'react';
+import { memo, useEffect, useState } from 'react';
+import { useWindowSize } from '../../../lib/utils';
```

Comment:
nit: absolute imports (for better auto-fixing of imports when we move files and avoiding lots of confusing `../`) for anything that's not in the same folder already